deepsensor/model/model.py

```diff
@@ -648,6 +648,597 @@ def unnormalise_pred_array(arr, **kwargs):
+                            / 2
+                        )
+                    )
+                )
+            else:
+                x2_overlap_index = int(
+                    np.floor(
+                        (
+                            X_t_ds.coords[orig_x2_name].values.size
+                            - int(
+                                np.ceil(
+                                    (
+                                        np.argmin(
+                                            np.abs(
+                                                X_t_ds.coords[orig_x2_name].values
+                                                - unnorm_overlap_x2
+                                            )
+                                        )
+                                    )
+                                )
+                            )
+                        )
+                        / 2
+                    )
+                )
+
+            x1_x2_overlap = (x1_overlap_index, x2_overlap_index)
+
+            return x1_x2_overlap
+
+        def get_index(*args, x1=True) -> Union[int, Tuple[List[int], List[int]]]:
+            """Convert coordinates into pixel row/column (index).
+
+            Parameters
+            ----------
+            args : tuple
+                If one argument (numeric), it represents the coordinate value.
+                If two arguments (lists), they represent lists of coordinate values.
+
+            x1 : bool, optional
+                If True, compute index for x1 (default is True).
+
+            Returns:
+            -------
+                Union[int, Tuple[List[int], List[int]]]
+                If one argument is provided and x1 is True or False, returns the index position.
+                If two arguments are provided, returns a tuple containing two lists:
+                - First list: indices corresponding to x1 coordinates.
+                - Second list: indices corresponding to x2 coordinates.
+
+            """
+            if len(args) == 1:
+                patch_coord = args
+                if x1:
+                    coord_index = np.argmin(
+                        np.abs(X_t.coords[orig_x1_name].values - patch_coord)
+                    )
+                else:
+                    coord_index = np.argmin(
+                        np.abs(X_t.coords[orig_x2_name].values - patch_coord)
+                    )
+                return coord_index
+
+            elif len(args) == 2:
+                patch_x1, patch_x2 = args
+                x1_index = [
+                    np.argmin(np.abs(X_t.coords[orig_x1_name].values - target_x1))
+                    for target_x1 in patch_x1
+                ]
+                x2_index = [
+                    np.argmin(np.abs(X_t.coords[orig_x2_name].values - target_x2))
+                    for target_x2 in patch_x2
+                ]
+                return (x1_index, x2_index)
+
+        def stitch_clipped_predictions(
```

Comment:
It would probably make more sense to put this method in deepsensor.model.pred, since it does not use `self`, and just operates on Prediction objects. WDYT?

Reply:
Thinking more on this, there are a number of methods specifically for stitching patches together, constituting a few hundred lines of code. A few options for organising are: moving into a specific module as functions; moving into the `Prediction` class as static methods; or moving into a child class e.g. `PatchwisePrediction(Prediction)`; moving to the `pred` module as functions outside of the `Prediction` class. Which would you prefer?